src/scheduler.ts

```diff
@@ -1,174 +1,247 @@
 import { WsProvider, ApiPromise } from '@polkadot/api'
-import { SubmittableExtrinsic, SubmittableResultSubscription } from '@polkadot/api/types'
-import { web3FromAddress } from '@polkadot/extension-dapp'
+import { Signer, SubmittableExtrinsic } from '@polkadot/api/types'
 import { Balance } from '@polkadot/types/interfaces'
 import { ISubmittableResult } from '@polkadot/types/types'
 import { HexString } from '@polkadot/util/types'
-import _ from 'lodash'
+import * as _ from 'lodash'
 
-export default class Scheduler {
+import {
+  LOWEST_TRANSFERRABLE_AMOUNT,
+  MIN_IN_HOUR,
+  MS_IN_SEC,
+  OakChainSchedulingLimit,
+  OakChainWebsockets,
+  RECURRING_TASK_LIMIT,
+  SEC_IN_MIN,
+} from './constants'
+
+export class Scheduler {
   wsProvider: WsProvider
   api: ApiPromise
+  chain: OakChains
+  schedulingTimeLimit: number
 
-  constructor(websocket: string) {
-    this.wsProvider = new WsProvider(websocket)
+  constructor(chain: OakChains) {
+    this.chain = chain
+    this.wsProvider = new WsProvider(OakChainWebsockets[chain])
+    this.schedulingTimeLimit = OakChainSchedulingLimit[chain]
   }
 
   private async getAPIClient(): Promise<ApiPromise> {
     if (_.isNil(this.api)) {
-      this.api = await ApiPromise.create({ provider: this.wsProvider })
+      this.api = await ApiPromise.create({
+        provider: this.wsProvider,
+        rpc: {
+          automationTime: {
+            generateTaskId: {
+              description: 'Getting task ID given account ID and provided ID',
+              params: [
+                {
+                  name: 'accountId',
+                  type: 'AccountId',
+                },
+                {
+                  name: 'providedId',
+                  type: 'Text',
+                },
+              ],
+              type: 'Hash',
+            },
+          },
+        },
+      })
     }
     return this.api
   }
 
-  private async getNonce(address: string): Promise<number> {
-    const api = await this.getAPIClient()
-    let fromNonceCodecIndex = await api.rpc.system.accountNextIndex(address)
-    return fromNonceCodecIndex.toNumber()
+  private convertToSeconds(startTimestamps: number[]): number[] {
+    return _.map(startTimestamps, (startTimestamp) => {
+      const isMillisecond = startTimestamp > 100000000000
```

Comment:
determined this with testing on https://www.unixtimestamp.com/index.php

100000000000 is the threshhold that they test on for second vs millisecond. for context, 99999999999 (the above number - 1) is Wed Nov 16 5138 09:46:39 GMT+0000, so i think we're safe for now